library/common/network/configurator.cc

```diff
@@ -60,21 +61,42 @@ namespace {
 #define SUPPORTS_GETIFADDRS
 #endif
 
-std::atomic<envoy_network_t> MobileUtility::preferred_network_{ENVOY_NET_GENERIC};
+SINGLETON_MANAGER_REGISTRATION(network_configurator);
 
-void MobileUtility::setPreferredNetwork(envoy_network_t network) { preferred_network_ = network; }
+constexpr absl::string_view BaseDnsCache = "base_dns_cache";
```

Comment:
I wonder if we can tie this somehow to the config being loaded so that we don't have the possibility of the names mismatching.

Reply:
Well, that's why I made the handling optional/nullable. I also anticipated that we'd eventually be doing this per-interface, so I didn't want to overthink it.

Reply:
sg. As long as we fire an event if this is null we should be able to observe if we are not getting a cache we are expecting.